litestar/response/base.py

```diff
@@ -218,128 +339,83 @@ def delete_cookie(
+            raise ImproperlyConfiguredException(f"unsupported media_type {media_type} for content {content!r}")
         except (AttributeError, ValueError, TypeError) as e:
             raise ImproperlyConfiguredException("Unable to serialize response content") from e
 
-    @property
-    def content_length(self) -> int:
-        """Content length of the response if applicable.
-
-        Returns:
-            The content length of the body (e.g. for use in a ``Content-Length`` header).
-            If the response does not have a body, this value is ``None``
-        """
-        if self.status_allows_body:
-            return len(self.body)
-        return 0
-
-    def encode_headers(self) -> list[tuple[bytes, bytes]]:
-        """Encode the response headers as a list of byte tuples.
-
-        Notes:
-            - A ``Content-Length`` header will be added if appropriate and not provided by the user.
-
-        Returns:
-            A list of tuples containing the headers and cookies of the request in a format ready for ASGI transmission.
-        """
-
-        return list(
-            chain(
-                ((k.lower().encode("latin-1"), str(v).encode("latin-1")) for k, v in self.headers.items()),
-                (cookie.to_encoded_header() for cookie in self.cookies),
-                self.raw_headers,
-            )
-        )
-
-    async def after_response(self) -> None:
-        """Execute after the response is sent.
-
-        Returns:
-            None
-        """
-        if self.background is not None:
-            await self.background()
-
-    async def start_response(self, send: Send) -> None:
-        """Emit the start event of the response. This event includes the headers and status codes.
-
-        Args:
-            send: The ASGI send function.
-
-        Returns:
-            None
-        """
-
-        encoded_headers = self.encode_headers()
-
-        content_length = self.content_length
-        if "content-length" not in self.headers and content_length:
-            encoded_headers.append((b"content-length", str(content_length).encode("latin-1")))
-
-        event: HTTPResponseStartEvent = {
-            "type": "http.response.start",
-            "status": self.status_code,
-            "headers": encoded_headers,
-        }
-
-        await send(event)
-
-    async def send_body(self, send: Send, receive: Receive) -> None:
-        """Emit the response body.
+    def to_asgi_response(
+        self,
+        *,
+        app: Litestar,
+        background: BackgroundTask | BackgroundTasks | None,
+        cookies: list[Cookie] | None,
+        encoded_headers: list[tuple[bytes, bytes]] | None,
+        headers: dict[str, str] | None,
+        is_head_response: bool,
+        media_type: MediaType | str | None,
+        request: Request,
+        status_code: int | None,
+        type_encoders: TypeEncodersMap | None,
+    ) -> ASGIResponse:
```

Comment:
Might make `app` and `request` positional and the rest kwargs with default values. It would make it easier to use this method downstream.